components/kyma-environment-broker/internal/process/provisioning/input/builder.go

```diff
@@ -76,18 +95,35 @@ func (f *InputBuilderFactory) ForPlan(planID string) (internal.ProvisionInputCre
 		mutex:                     nsync.NewNamedMutex(),
 		hyperscalerInputProvider:  provider,
 		optionalComponentsService: f.optComponentsSvc,
-	}, true
+	}, nil
 }
 
-func (f *InputBuilderFactory) initInput(provider HyperscalerInputProvider) gqlschema.ProvisionRuntimeInput {
+func (f *InputBuilderFactory) initInput(provider HyperscalerInputProvider, kymaVersion string) (gqlschema.ProvisionRuntimeInput, error) {
+	var (
+		version    string
+		components internal.ComponentConfigurationInputList
+	)
+
+	if kymaVersion != "" {
```

Comment:
if you already implemented the "cache" mechanism for `componentsProvider.AllComponents` maybe this function can be simplified as this method can be always executed with a proper Kyma version.

Reply:
First of all, I call `AllComponents` method in InpuBuilder constructor, it's made to be sure that components from the default version will always be present (if this call failed then all Pod will be failed).
It would look weird if I saved the default component list as a struct field and then always call about components to external service. It would look weird from the point of view of the builder's logic.
Second, InpuBuilder doesn't know what exactly is inside `AllComponents`, if there is cache or not.
I'd leave it the way it is